src/test/database/remove.spec.ts

```diff
@@ -0,0 +1,196 @@
+"use strict";
+
+import { expect } from "chai";
+import * as pathLib from "path";
+
+import DatabaseRemove from "../../database/remove";
+
+class TestRemote {
+  public data: any;
+
+  constructor(data: any) {
+    this.data = data;
+  }
+
+  public deletePath(path: string): Promise<boolean> {
+    if (path === "/") {
+      this.data = null;
+      return Promise.resolve(true);
+    }
+    const parentDir = pathLib.dirname(path);
+    const basename = pathLib.basename(path);
+    delete this._dataAtpath(parentDir)[basename];
+    if (Object.keys(this._dataAtpath(parentDir)).length === 0) {
+      return this.deletePath(parentDir);
+    }
+    return Promise.resolve(true);
+  }
+
+  public prefetchTest(path: string): Promise<string> {
+    const d = this._dataAtpath(path);
+    if (!d) {
+      return Promise.resolve("empty");
+    }
+    if ("string" === typeof d) {
+      return Promise.resolve("small");
+    } else if (Object.keys(d).length === 0) {
+      return Promise.resolve("empty");
+    } else {
+      return Promise.resolve("large");
+    }
+  }
+
+  public listPath(path: string): Promise<string[]> {
+    const d = this._dataAtpath(path);
+    if (d) {
+      return Promise.resolve(Object.keys(d));
+    }
+    return Promise.resolve([]);
+  }
+
+  private _dataAtpath(path: string): any {
+    const splitedPath = path.slice(1).split("/");
+    let d = this.data;
+    for (const p of splitedPath) {
+      if (d && p !== "") {
+        if ("string" === typeof d) {
```

Comment:
This is something called "yoda conditionals" - it's just backwards. I see one above that I missed as well - the same applies.

```suggestion
        if (typeof d === "string") {
```